server/src/server.js

```diff
@@ -1,337 +1,189 @@
-let http = require("http"),
-  path = require('path'),
-  url = require('url'),
-  fs = require('fs'),
-  React = require("react"),
-  Router = require("react-router"),
-  routes = require("./routes.js"),
-  models = require("./models.js"),
-  lookupContentType = require('mime-types').contentType,
-  git = require('git-rev'),
-  Cookies = require("cookies"),
-  qs = require('querystring');
+const path = require('path');
+const Cookies = require("cookies");
 
 const { Shot } = require("./servershot");
 const { checkLogin, registerLogin } = require("./users");
 const dbschema = require("./dbschema");
+const express = require("express");
+const bodyParser = require('body-parser');
+const morgan = require("morgan");
+const linker = require("./linker");
 
-const jspath = "/js/",
-  csspath = "/css/",
-  modelspath = "/data/",
-  contentpath = "/content/",
-  imgpath = "/img/",
-  favicopath = "/favicon.ico",
-  loginpath = "/api/login",
-  registerpath = "/api/register",
-  clipImagePath = "/clip/",
-  contentType = "Content-type",
-  notFound = "Not Found",
-  doctype = "<!DOCTYPE html>",
-  footer = "</body></html>",
-  script = `
-function gotData(Handler, data) {
-  data.linkify = linkify;
-  routes.setGitRevision(data.gitRevision);
-  React.render(React.createElement(Handler, data), document);
-}
+const contentType = "Content-type";
 
-Router.run(routes.routes, Router.HistoryLocation, function (Handler, state) {
-  if (cachedData) {
-    var _d = cachedData;
-    cachedData = null;
-    gotData(Handler, _d);
-    return;
-  }
+dbschema.createTables();
+dbschema.createKeygrip();
 
-  var appnames = [],
-    app = null;
-  for (var i in state.routes) {
-    if (!!state.routes[i].name) {
-      appnames.push(state.routes[i].name);
-      if (app === null) {
-        app = state.routes[i];
-      }
-    }
-  }
-  if (!appnames.length) {
-    console.error("Error: No app was routed");
-    return;
-  }
+const app = express();
 
-  console.log("Route to app", appnames[0]);
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(bodyParser.json({limit: '100mb'}));
 
-  if (appnames[0] === "shot") {
-    var key = state.params.shotId + "/" + state.params.shotDomain,
-      xhr = new XMLHttpRequest();
+app.use("/static", express.static(path.join(__dirname, "static"), {
+  index: false
+}));
 
-    xhr.onload = function () {
-      if (xhr.status === 200) {
-        var data = JSON.parse(xhr.responseText);
-        gotData(Handler, {backend: location.origin, id: key, shot: data});
-      } else {
-        console.error("Error: Bad response: ", xhr.status, xhr.responseText);
-      }
-    };
+app.use(morgan("dev"));
 
-    xhr.open("GET", "/data/" + key);
-    xhr.send();
-  }
+app.use(function (req, res, next) {
+  let cookies = new Cookies(req, res, dbschema.getKeygrip());
+  req.userId = cookies.get("user", {signed: true});
+  let proto = req.headers["x-forwarded-proto"] || "http";
```

Comment:
Alternatively, you can use `app.set('trust proxy', true)`, and then this becomes `req.protocol`. It'll also set `req.ip` based on the `X-Forwarded-For` header.